presto-jdbc/src/main/java/com/facebook/presto/jdbc/QueryExecutor.java

```diff
@@ -94,7 +90,19 @@ protected void finalize()
 
     static QueryExecutor create(String userAgent)
     {
-        return new QueryExecutor(userAgent, jsonCodec(QueryResults.class), jsonCodec(ServerInfo.class), getSystemSocksProxy());
+        JettyHttpClient httpClient = new JettyHttpClient(
+                new HttpClientConfig()
+                        .setConnectTimeout(new Duration(10, TimeUnit.SECONDS))
+                        .setSocksProxy(getSystemSocksProxy()),
+                new JettyIoPool("presto-jdbc", new JettyIoPoolConfig()),
+                ImmutableSet.of(new UserAgentRequestFilter(userAgent)));
+
+        return new QueryExecutor(userAgent, jsonCodec(QueryResults.class), jsonCodec(ServerInfo.class), httpClient);
+    }
+
+    static QueryExecutor createTestExecutor(String userAgent, HttpClient testHttpClient)
```

Comment:
Name this variable `httpClient` since "test" is a detail of the caller (this class doesn't care what it is)